src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -271,24 +269,27 @@ def sync_deploy_root_with_stage(self, role: str) -> DiffResult:
             )
 
         # Perform a diff operation and display results to the user for informational purposes
-        log.info(
-            'Performing a diff between the Snowflake stage and your local deploy_root ("%s") directory.',
-            self.deploy_root,
+        cc.step(
+            "Performing a diff between the Snowflake stage and your local deploy_root ('%s') directory."
+            % self.deploy_root
         )
         diff: DiffResult = stage_diff(self.deploy_root, self.stage_fqn)
-        log.info("Listing results of diff:")
-        log.info("New files only on your local: %s", ",".join(diff.only_local))
-        log.info("New files only on the stage: %s", ",".join(diff.only_on_stage))
-        log.info(
-            "Existing files modified or status unknown: %s", ",".join(diff.different)
-        )
-        log.info("Existing files identical to the stage: %s", ",".join(diff.identical))
+        with cc.phase("Listing results of diff:"):
```

Comment:
Not sure that's a phase. Conceptually, a phase should be a group of related steps. We need to be careful since phases can't be nested, so we most likely only want to create them directly within top level commands, and not in helper code. We should definitely avoid using phase/step to control indentation. Maybe the phase here is the deploy phase within run?

The console API is pretty minimal, I think we'll need to revisit it when we add `snow app status`. Until then, I'd be OK just performing any required indentation in each step directly.